### DISCUSSION TITLE
#include nlohmann/json in header file causes "Cannot open source file" error

### DISCUSSION THREAD

**Discussion by chinloong93** [nlohmann/json#4351] [Q&A]
I'm using CMake with VisualStudio 2022 to build my project.

My CMake file settings look like the following:

```
FetchContent_Declare(
	json 
	URL https://github.com/nlohmann/json/releases/download/v3.11.3/json.tar.xz
)
set(JSON_BuildTests OFF CACHE INTERNAL "")
FetchContent_MakeAvailable(json)

add_library(counter_config counter_config.cc)
target_link_libraries(
	counter_config
	PUBLIC nlohmann_json::nlohmann_json
)
```
When I `#include <nlohmann/json>` into the header file (.h), I get a "Cannot open source file" error. However, this does not occur when I include it into a source file (.cpp). Is there an explanation for this? 

Thank you!

**Comment by nlohmann** [comment]
Very strange: the same setup works for me.

**Reply by chinloong93** [reply]
I agree. Extremely strange. Thanks for checking! Will do more investigation on my end

**Reply by jar3n** [reply]
Has there been an update on this? I am trying a similar thing with the latest version 3.12.0 and cmake cant find the json.hpp when i try to use #include <nlohmann/json.hpp>. I am hoping to use the json in a library because that is how my code is structured.
